frontend/src/Pages/Error.tsx

```diff
@@ -1,23 +1,75 @@
 import { AUTHCALLBACK } from '@/useAuth';
+import { ErrorType } from '@/common';
+import { useNavigate } from 'react-router-dom';
+
+interface ErrorPageProps {
+    type?: ErrorType;
+    back?: boolean;
+    message?: string;
+    navigateTo?: string;
+}
+
+const errorMap: Record<
+    ErrorType,
+    { message: string; buttonText: string; onClick: () => void }
+> = {
+    unauthorized: {
+        message:
+            'You do not have permission to access this page. Contact your administrator if you believe you have reached this page in error.',
+        buttonText: 'Home Page',
+        onClick: () => (window.location.href = AUTHCALLBACK)
+    },
+    'not-found': {
+        message: 'The page you requested does not exist.',
+        buttonText: 'Home Page',
+        onClick: () => (window.location.href = AUTHCALLBACK)
+    },
+    'server-error': {
+        message: 'An unexpected error occurred. Please try again later.',
+        buttonText: 'Home Page',
+        onClick: () => (window.location.href = AUTHCALLBACK)
+    },
+    'auth-required': {
```

Comment:
@CK-7vn Can we remove the auth-required option? The preferred outcome of an unauthenticated user trying to access a page behind authentication is that they get redirected to the auth page automatically.